chore(wallet)_: estimated time improvements

saledjenic opened the pull request:
- deduplicated base fee and priority fee array before processing
- removed network congestion index, cause previously the number of blocks fetched for the calculation was reduced and after deduplication, it's not needed
- check priority fee for L1 chains only

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (56)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | d5293b98 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6369/1/) | 2025-02-26 15:14:17 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250226-151112-d5293b-pr6369.zip) |
| :heavy_check_mark: | d5293b98 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6369/1/) | 2025-02-26 15:14:26 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250226-151111-d5293b-pr6369.aar) |
| :heavy_multiplication_x: | d5293b98 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/1/) | 2025-02-26 15:15:31 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/1//consoleText) |
| :heavy_check_mark: | d5293b98 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6369/1/) | 2025-02-26 15:15:49 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250226-151115-d5293b-pr6369.zip) |
| :heavy_check_mark: | d5293b98 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6369/1/) | 2025-02-26 15:16:36 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250226-151116-d5293b-pr6369.zip) |
| :heavy_check_mark: | d5293b98 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6369/1/) | 2025-02-26 15:16:50 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250226-151121-d5293b-pr6369.zip) |
| :heavy_check_mark: | d5293b98 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6369/1/) | 2025-02-26 15:17:11 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250226-151114-d5293b-pr6369.zip) |
| :heavy_check_mark: | d5293b98 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/1/) | 2025-02-26 15:25:15 | ~13 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ef9965fc | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6369/2/) | 2025-03-24 07:46:35 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250324-074329-ef9965-pr6369.aar) |
| :heavy_check_mark: | ef9965fc | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6369/2/) | 2025-03-24 07:46:35 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250324-074330-ef9965-pr6369.zip) |
| :heavy_multiplication_x: | ef9965fc | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/2/) | 2025-03-24 07:47:57 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/2//consoleText) |
| :heavy_check_mark: | ef9965fc | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6369/2/) | 2025-03-24 07:48:04 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250324-074332-ef9965-pr6369.zip) |
| :heavy_check_mark: | ef9965fc | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6369/2/) | 2025-03-24 07:48:49 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250324-074334-ef9965-pr6369.zip) |
| :heavy_check_mark: | ef9965fc | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6369/2/) | 2025-03-24 07:49:01 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250324-074337-ef9965-pr6369.zip) |
| :heavy_check_mark: | ef9965fc | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6369/2/) | 2025-03-24 07:49:24 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250324-074331-ef9965-pr6369.zip) |
| :heavy_check_mark: | ef9965fc | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/2/) | 2025-03-24 08:14:21 | ~30 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 6798ffa5 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6369/3/) | 2025-03-26 07:20:53 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250326-071754-6798ff-pr6369.aar) |
| :heavy_check_mark: | 6798ffa5 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6369/3/) | 2025-03-26 07:20:56 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250326-071755-6798ff-pr6369.zip) |
| :heavy_multiplication_x: | 6798ffa5 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/3/) | 2025-03-26 07:21:27 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/3//consoleText) |
| :heavy_check_mark: | 6798ffa5 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6369/3/) | 2025-03-26 07:22:18 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250326-071802-6798ff-pr6369.zip) |
| :heavy_check_mark: | 6798ffa5 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6369/3/) | 2025-03-26 07:22:30 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250326-071758-6798ff-pr6369.zip) |
| :heavy_check_mark: | 6798ffa5 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6369/3/) | 2025-03-26 07:23:06 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250326-071758-6798ff-pr6369.zip) |
| :heavy_check_mark: | 6798ffa5 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6369/3/) | 2025-03-26 07:23:43 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250326-071756-6798ff-pr6369.zip) |
| :heavy_check_mark: | 6798ffa5 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/3/) | 2025-03-26 07:47:53 | ~29 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 55dbe031 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6369/4/) | 2025-03-26 13:20:37 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250326-131757-55dbe0-pr6369.zip) |
| :heavy_check_mark: | 55dbe031 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6369/4/) | 2025-03-26 13:21:13 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250326-131756-55dbe0-pr6369.aar) |
| :heavy_multiplication_x: | 55dbe031 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/4/) | 2025-03-26 13:21:28 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/4//consoleText) |
| :heavy_check_mark: | 55dbe031 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6369/4/) | 2025-03-26 13:21:57 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250326-131759-55dbe0-pr6369.zip) |
| :heavy_check_mark: | 55dbe031 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6369/4/) | 2025-03-26 13:22:33 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250326-131806-55dbe0-pr6369.zip) |
| :heavy_check_mark: | 55dbe031 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6369/4/) | 2025-03-26 13:23:10 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250326-131801-55dbe0-pr6369.zip) |
| :heavy_check_mark: | 55dbe031 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6369/4/) | 2025-03-26 13:25:01 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250326-131758-55dbe0-pr6369.zip) |
| :heavy_multiplication_x: | 55dbe031 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/4/) | 2025-03-26 13:50:03 | ~31 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | bcb4f458 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6369/5/) | 2025-03-26 15:14:19 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250326-151117-bcb4f4-pr6369.aar) |
| :heavy_check_mark: | bcb4f458 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6369/5/) | 2025-03-26 15:14:29 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250326-151119-bcb4f4-pr6369.zip) |
| :heavy_multiplication_x: | bcb4f458 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/5/) | 2025-03-26 15:15:35 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/5//consoleText) |
| :heavy_check_mark: | bcb4f458 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6369/5/) | 2025-03-26 15:15:37 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250326-151127-bcb4f4-pr6369.zip) |
| :heavy_check_mark: | bcb4f458 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6369/5/) | 2025-03-26 15:16:18 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250326-151122-bcb4f4-pr6369.zip) |
| :heavy_check_mark: | bcb4f458 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6369/5/) | 2025-03-26 15:17:18 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250326-151120-bcb4f4-pr6369.zip) |
| :heavy_check_mark: | bcb4f458 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6369/5/) | 2025-03-26 15:17:23 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250326-151123-bcb4f4-pr6369.zip) |
| :heavy_multiplication_x: | bcb4f458 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/5/) | 2025-03-26 15:42:21 | ~30 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 647b3903 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6369/6/) | 2025-03-26 15:55:50 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250326-155255-647b39-pr6369.zip) |
| :heavy_check_mark: | 647b3903 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6369/6/) | 2025-03-26 15:55:52 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250326-155255-647b39-pr6369.aar) |
| :heavy_check_mark: | 647b3903 | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6369/6/) | 2025-03-26 15:57:05 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250326-155306-647b39-pr6369.zip) |
| :heavy_check_mark: | 647b3903 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6369/6/) | 2025-03-26 15:57:31 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250326-155259-647b39-pr6369.zip) |
| :heavy_check_mark: | 647b3903 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6369/6/) | 2025-03-26 15:58:10 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250326-155302-647b39-pr6369.zip) |
| :heavy_check_mark: | 647b3903 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6369/6/) | 2025-03-26 15:58:47 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250326-155258-647b39-pr6369.zip) |
| :heavy_check_mark: | 647b3903 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/6/) | 2025-03-26 16:22:56 | ~29 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/6//consoleText) |
| :heavy_check_mark: | 647b3903 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/6/) | 2025-03-26 16:27:47 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4a62cd32 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6369/7/) | 2025-03-31 16:07:07 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250331-160407-4a62cd-pr6369.aar) |
| :heavy_check_mark: | 4a62cd32 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6369/7/) | 2025-03-31 16:07:20 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250331-160408-4a62cd-pr6369.zip) |
| :heavy_check_mark: | 4a62cd32 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6369/7/) | 2025-03-31 16:08:31 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250331-160417-4a62cd-pr6369.zip) |
| :heavy_check_mark: | 4a62cd32 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6369/7/) | 2025-03-31 16:08:41 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250331-160412-4a62cd-pr6369.zip) |
| :heavy_check_mark: | 4a62cd32 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6369/7/) | 2025-03-31 16:09:19 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250331-160414-4a62cd-pr6369.zip) |
| :heavy_check_mark: | 4a62cd32 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6369/7/) | 2025-03-31 16:09:57 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250331-160410-4a62cd-pr6369.zip) |
| :heavy_check_mark: | 4a62cd32 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/7/) | 2025-03-31 16:13:07 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/7//consoleText) |
| :heavy_multiplication_x: | 4a62cd32 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/7/) | 2025-03-31 16:40:53 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/7//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 9210a6c5 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6369/8/) | 2025-03-31 16:14:07 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250331-161113-9210a6-pr6369.zip) |
| :heavy_check_mark: | 9210a6c5 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6369/8/) | 2025-03-31 16:14:12 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250331-161112-9210a6-pr6369.aar) |
| :heavy_check_mark: | 9210a6c5 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6369/8/) | 2025-03-31 16:15:29 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250331-161123-9210a6-pr6369.zip) |
| :heavy_check_mark: | 9210a6c5 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6369/8/) | 2025-03-31 16:15:55 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250331-161117-9210a6-pr6369.zip) |
| :heavy_check_mark: | 9210a6c5 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6369/8/) | 2025-03-31 16:16:27 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250331-161119-9210a6-pr6369.zip) |
| :heavy_check_mark: | 9210a6c5 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6369/8/) | 2025-03-31 16:17:00 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250331-161114-9210a6-pr6369.zip) |
| :heavy_check_mark: | 9210a6c5 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/8/) | 2025-03-31 16:21:06 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/8//consoleText) |
| :heavy_multiplication_x: | 9210a6c5 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/8/) | 2025-03-31 17:17:01 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 277bed75 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6369/9/) | 2025-04-01 06:35:52 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250401-063255-277bed-pr6369.zip) |
| :heavy_check_mark: | 277bed75 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6369/9/) | 2025-04-01 06:36:03 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250401-063254-277bed-pr6369.aar) |
| :heavy_check_mark: | 277bed75 | [#9](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6369/9/) | 2025-04-01 06:37:03 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250401-063304-277bed-pr6369.zip) |
| :heavy_check_mark: | 277bed75 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6369/9/) | 2025-04-01 06:37:30 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250401-063300-277bed-pr6369.zip) |
| :heavy_check_mark: | 277bed75 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6369/9/) | 2025-04-01 06:38:06 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250401-063300-277bed-pr6369.zip) |
| :heavy_check_mark: | 277bed75 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6369/9/) | 2025-04-01 06:38:49 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250401-063257-277bed-pr6369.zip) |
| :heavy_check_mark: | 277bed75 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/9/) | 2025-04-01 06:41:43 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6369/9//consoleText) |
| :heavy_multiplication_x: | 277bed75 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/9/) | 2025-04-01 07:08:31 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6369/9//consoleText) |

saledjenic commented:
We should think more about estimation, this approach is a temporary solution.

saledjenic commented on services/wallet/router/fees/estimated_time.go:
```diff
@@ -215,8 +233,8 @@ func (f *FeeManager) estimatedTimeV2(feeHistory *FeeHistory, txMaxFeePerGas *big
-		baseFeePercentileIndex := getBaseFeePercentileIndex(sortedBaseFees, p.baseFeePercentile, networkCongestion)
-		if txBaseFee.Cmp(sortedBaseFees[baseFeePercentileIndex]) >= 0 && txPriorityFee.Cmp(p.priorityFee) >= 0 {
+		baseFeePercentileIndex := getBaseFeePercentileIndex(uniqueBaseFees, p.baseFeePercentile, networkCongestion)
+		if txBaseFee.Cmp(uniqueBaseFees[baseFeePercentileIndex]) >= 0 && txPriorityFee.Cmp(p.priorityFee) >= 0 {
```
I am not sure if we should check this `txPriorityFee.Cmp(p.priorityFee) >= 0` in this condition?
Or should we check it for all chains?

saledjenic replied:
I've updated this condition to check for priority fees only if estimating time for L1 chains.

saledjenic commented:
I guess that with the changes done here "Unknown" (0) estimated time will be displayed very rarely.

saledjenic committed:
chore(wallet)_: estimated time improvements

- deduplicated base fee and priority fee array before processing
- removed network congestion index, cause previously the number of blocks fetched
for the calculation was reduced and after deduplication it's not needed
- check priority fee for L1 chains only
